### PR TITLE
Introduce new Promise API methods

### PR DESCRIPTION
**Pull request by galpeter**
The new API methods make it possible to get a Promise object's
result and it's state.

### CODE REVIEW

**Comment by galpeter** [comment]
Documentation will be added a bit later. First I would like to get a feedback on the API.

**Comment by rerobika** [comment]
I support this idea! +1

**Comment by dbatyai** [comment]
The new functions look reasonable, I support this as well.

**Comment by galpeter** [comment]
I've rebased and updated the PR with docs.

File: docs/02.API-REFERENCE.md [hunk 1]
```diff
@@ -595,6 +595,22 @@ typedef jerry_value_t (*jerry_vm_exec_stop_callback_t) (void *user_p);
 
 - [jerry_set_vm_exec_stop_callback](#jerry_set_vm_exec_stop_callback)
 
+## jerry_promise_state_t
+
+Enum which describes the state of a Promise.
+
+Possible values:
+
+ - JERRY_PROMISE_STATE_NONE - Invalid/Unkown state (possibly called on a non-promise object).
```
Comment (rerobika) [inline docs/02.API-REFERENCE.md]: Typo: `Unkown`

File: jerry-core/api/jerry.c [hunk 2]
```diff
@@ -3078,6 +3086,57 @@ jerry_resolve_or_reject_promise (jerry_value_t promise, /**< the promise value *
...
+jerry_value_t
+jerry_get_promise_result (const jerry_value_t promise) /**< promise object to get the result from */
+{
+#if ENABLED (JERRY_ES2015_BUILTIN_PROMISE)
+  if (!jerry_value_is_promise (promise))
+  {
+    return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG (wrong_args_msg_p)));
+  }
+
+  ecma_value_t result = ecma_promise_get_result (ecma_get_object_from_value (promise));
```
Comment (rerobika) [inline jerry-core/api/jerry.c]: Cannot we return directly?

Comment (galpeter) [reply jerry-core/api/jerry.c]: Yes, I've updated the PR with the fixes.

**Pull request merged**